From 42ee0054316e9175af11a3e9a380fc7a21095cfa Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:06:26 -0400
Subject: [tools] perf: Remove extra '/' character in events file path

Message-id: <1410181752-25631-142-git-send-email-jolsa@redhat.com>
Patchwork-id: 91823
O-Subject: [PATCH RHEL7.1 BZ1134356 141/307] perf tools: Remove extra '/' character in events file path
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 603940b6b8b1347ec13a628165c00194ebc17ed2
Author: Xia Kaixu <xiakaixu@huawei.com>
Date: Sat Apr 26 15:55:12 2014 +0800

description
===========
The array debugfs_known_mountpoints[] will cause extra '/'
character output.
Remove it.

pre:
$ perf probe -l
/sys/kernel/debug//tracing/uprobe_events file does not exist -
please rebuild kernel with CONFIG_UPROBE_EVENTS.

post:
$ perf probe -l
/sys/kernel/debug/tracing/uprobe_events file does not exist -
please rebuild kernel with CONFIG_UPROBE_EVENTS.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/api/fs/debugfs.c b/tools/lib/api/fs/debugfs.c
index 7c43479..a74fba6 100644
--- a/tools/lib/api/fs/debugfs.c
+++ b/tools/lib/api/fs/debugfs.c
@@ -12,8 +12,8 @@
 char debugfs_mountpoint[PATH_MAX + 1] = "/sys/kernel/debug";
 
 static const char * const debugfs_known_mountpoints[] = {
- "/sys/kernel/debug/",
- "/debug/",
+ "/sys/kernel/debug",
+ "/debug",
  0,
 };
 
-- 
1.7.1